cmd/editor_config.go

```diff
@@ -0,0 +1,140 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	u "github.com/cloudposse/atmos/pkg/utils"
+
+	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/config"
+	er "github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error"
+	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files"
+	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/utils"
+	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation"
+	"github.com/spf13/cobra"
+)
+
+var (
+	editorConfigVersion   = "v3.0.0"
+	defaultConfigFilePath = ".ecrc"
+	currentConfig         *config.Config
+	cliConfig             config.Config
+	configFilePath        string
+	tmpExclude            string
+)
+
+var editorConfigCmd *cobra.Command = &cobra.Command{
+	Use:   "editorconfig-checker",
+	Short: "EditorConfig Checker CLI",
+	Long:  "A command-line tool to check your files against the EditorConfig rules",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		initializeConfig()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		runMainLogic()
+	},
+}
+
+// initializeConfig breaks the initialization cycle by separating the config setup
+func initializeConfig() {
+	u.LogInfo(atmosConfig, fmt.Sprintf("EditorConfig Checker CLI Version: %s", editorConfigVersion))
+	if configFilePath == "" {
+		configFilePath = defaultConfigFilePath
+	}
+
+	var err error
+	currentConfig, err = config.NewConfig(configFilePath)
+	if err != nil {
+		u.LogError(atmosConfig, err)
+		os.Exit(1)
+	}
+
+	_ = currentConfig.Parse()
```

Comment:
We would not be fixing this one because:
It is as per the code of editorconfig. I tried to understand why it is but i guess it is not worth looking into why they wrote it that way. Let us just keep it as is and update it if we find any issue. Currently i found it working perfect. I updated the tests as well